modules/imap/site.js

```diff
@@ -133,6 +133,10 @@ var imap_delete_message = function(state, supplied_uid, supplied_detail) {
             {'name': 'folder', 'value': detail.folder}],
             function(res) {
                 if (!res.imap_delete_error) {
+                    const listPath = getParam('list_parent') || getListPathParam();
+                    const store = new Hm_MessagesStore(listPath, Hm_Utils.get_url_page_number(), `${getParam('keyword')}_${getParam('filter')}`);
+                    store.load();
+                    store.removeRow(uid);
                     if (Hm_Utils.get_from_global('msg_uid', false)) {
                         return;
                     }
```

Comment:
@mercihabam This line and the other below should also be removed. Also remove remove_from_cached_imap_pages and check the condition above, if it is necessary then it should go after the changes. Thank you!